libraries/node-core-library/src/LockFile.ts

```diff
@@ -6,28 +6,105 @@ import * as path from 'path';
 import * as child_process from 'child_process';
 import { setTimeout } from 'timers';
 
+/**
+ * Parses the process start time from a linux /proc/1/stat file.
+ * @param stat The contents of a linux /proc/1/stat file.
+ * @returns The process start time in yiffies.
+ */
+export function getProcessStartTimeFromProcStat (stat: string): string | undefined {
+  // Parse value 22.
+  // We cannot just split stat on spaces, because value 2 may contain spaces.
+  // For example, when running the following Shell commands:
+  // > cp "$(which bash)" ./'bash 2)('
+  // > ./'bash 2)(' -c 'OWNPID=$BASHPID;cat /proc/$OWNPID/stat'
+  // 59389 (bash 2)() S 59358 59389 59358 34818 59389 4202496 329 0 0 0 0 0 0 0 20 0 1 0
+  // > rm -rf ./'bash 2)('
+  // The output shows a stat file such that value 2 contains spaces.
+  // To still umambiguously parse such output we assume all values after the third consist of only digits...
+
+  // trimRight to remove the trailing line terminator.
+  let values: string[] = stat.trimRight().split(' ');
+  let i: number = values.length - 1;
+  while (i >= 0 && /^[0-9]+$/.test(values[i])) {
+    i -= 1;
+  }
+  // i is the index of the third value (but i need not be 2).
+  if (i < 2) {
+    // Format of stat has changed.
+    return undefined;
+  }
+  const value2: string = values.slice(1, i - 1).join(' ');
+  values = [values[0], value2].concat(values.slice(i));
+  if (values.length < 22) {
+    // Older version of linux, or non-standard configuration of linux.
+    return undefined;
+  }
+  const startTimeYiffies: string = values[21];
+  // In theory, the representations of start time returned by /proc/*/stat and ps -o lstart can change while the
+  // system is running, but we assume this does not happen.
+  // So the caller can safely use this value as part of a unique process id (on the machine, without comparing
+  // accross reboots).
```

Comment:
There are a couple interesting points here:
- The `ps -o lstart` does print out a full date/time. If that actually did change while the system is running (e.g. due to a clock synchronization), that would be problematic for the algorithm, because it would incorrectly infer that a lock has expired.  If there was a way to improve that, it might be interesting to consider (in a separate GitHub issue)
- The `/proc/*/stat` seems to show a time relative to the kernel boot, which would get invalidated when the PC reboots. That seems pretty safe. The chances of a second process getting spawned after a reboot, with the exact same starttime and PID seems astronomically small.